lib/pf/services/manager/dhcpd.pm

```diff
@@ -47,6 +47,9 @@ sub generateConfig {
     $tags{'networks'} = '';
     $tags{'active'} = '';
 
+    my $captive_portal_uri = isenabled($Config{'captive_portal'}{'secure_redirect'}) ? $HTTPS : $HTTP;
+    $captive_portal_uri .= "://" . $Config{'general'}{'hostname'} . "." . $Config{'general'}{'domain'};
```

Comment:
I think it would be the time to create a helper for the portal URL in pf::config::util

![image](https://cloud.githubusercontent.com/assets/3857942/14016007/97b03b56-f193-11e5-89cb-f1c467cedbd0.png)